Gem/Code/Source/GameLift/GameLiftServerComponent.cpp

```diff
@@ -0,0 +1,211 @@
+        provided.push_back(AZ_CRC_CE("MultiplayerSampleGameLiftServerService"));
+    }
+
+    void GameLiftServerSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("MultiplayerSampleGameLiftServerService"));
+    }
+
+    void GameLiftServerSystemComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("NetworkingService"));
+        required.push_back(AZ_CRC_CE("MultiplayerService"));
+        required.push_back(AZ_CRC_CE("AWSGameLiftServerService"));
+    }
+
+    void GameLiftServerSystemComponent::GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+        AZ_UNUSED(dependent);
+    }
+
+    void GameLiftServerSystemComponent::Init()
+    {
+        m_backfillProcessTerminated = true;
+        m_backfillComplete = true;
+    }
+
+    void GameLiftServerSystemComponent::Activate()
+    {
+        AzFramework::SessionNotificationBus::Handler::BusConnect();
+
+        AWSGameLift::AWSGameLiftServerRequestBus::Broadcast(&AWSGameLift::AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady);
+    }
+
+    void GameLiftServerSystemComponent::Deactivate()
+    {
+        StopBackfillProcess();
+
+        AzFramework::SessionNotificationBus::Handler::BusDisconnect();
+    }
+
+    bool GameLiftServerSystemComponent::OnSessionHealthCheck()
+    {
+        return true;
+    }
+
+    bool GameLiftServerSystemComponent::OnCreateSessionBegin(const AzFramework::SessionConfig& sessionConfig)
+    {
+        m_maxPlayer = sessionConfig.m_maxPlayer;
+
+        return true;
+    }
+
+    void GameLiftServerSystemComponent::OnCreateSessionEnd()
+    {
+        StartBackfillProcess();
+    }
+
+    bool GameLiftServerSystemComponent::OnDestroySessionBegin()
+    {
+        StopBackfillProcess();
+
+        return true;
+    }
+
+    void GameLiftServerSystemComponent::OnDestroySessionEnd()
+    {
+        m_maxPlayer = 0;
+    }
+
+    void GameLiftServerSystemComponent::OnUpdateSessionBegin(
+        const AzFramework::SessionConfig& sessionConfig, const AZStd::string& updateReason)
+    {
+        AZ_UNUSED(sessionConfig);
+        if (updateReason == "BACKFILL_FAILED" || updateReason == "BACKFILL_TIMED_OUT" || updateReason == "BACKFILL_CANCELLED")
+        {
+            m_backfillComplete = true;
+            m_waitEvent.release();
+        }
+    }
+
+    void GameLiftServerSystemComponent::OnUpdateSessionEnd()
+    {
+        AZ_TracePrintf("MultiplayerSample", "Multiplayer sample server manual backfill has completed");
+
+        m_backfillComplete = true;
+    }
+
+    void GameLiftServerSystemComponent::StartBackfillProcess()
+    {
+        bool manualBackfill = false;
+        const auto console = AZ::Interface<AZ::IConsole>::Get();
+        if (console->GetCvarValue("sv_manualBackfill", manualBackfill) != AZ::GetValueResult::Success)
```

Comment:
You don't have to do any of this if the cvar is declared within the same translation unit. Just use sv_manualBackfill as you would any normal static variable. That AZ:CVAR macro will expand to something similar to `static bool sv_manualBackfill = false;`, just with some extra logic for console binding and thread safety.

That means this whole block can just be

```
void GameLiftServerSystemComponent::StartBackfillProcess()
    {
        if (!sv_manualBackfill|| !m_backfillProcessTerminated)
        {
            return;
        }
```